### ISSUE TITLE
Failed remove flow types from async function (generic)

### ISSUE THREAD

**Issue by salisbury-espinosa** [flowtype/flow-remove-types#76]
#### Source code: 
```js
/* @flow */

type ObjMessageType<T> = {
  message: T
};
const f = async <MessageType>(
  message: MessageType
): Promise<ObjMessageType<MessageType>> => {
   await Promise.resolve();
   return Promise.resolve({
    message
  });       
};
```

#### flow-check 
Ok
https://flow.org/try :  **No errors!**

#### flow-remove-types 
NOT Ok
```bash
flow-remove-types /tmp/test.types.js -d /tmp/build --all --pretty
/tmp/test.types.js
 ↳ Syntax Error: Unexpected token, expected ; (8:1)
   7:  ): Promise<ObjMessageType<MessageType>> => {
```

the error is reproduced only for **async** function!

**BUT**

```js
/* @flow */

type ObjMessageType<T> = {
  message: T
};
async function f <MessageType> (
  message: MessageType
): Promise<ObjMessageType<MessageType>> {
   await Promise.resolve();
   return Promise.resolve({
    message
  });       
};
```

it makes no mistakes neither by **flow-check**  nor by **flow-remove-types** 
#### flow-check 
Ok
https://flow.org/try :  **No errors!**
#### flow-remove-types 
Ok
```bash
flow-remove-types /tmp/test.types.js -d /tmp/build --all --pretty
/tmp/test.types.js
 ↳ /tmp/build/tmp/test.types.js
```

**Comment by motiz88** [comment]
This will be resolved in the next major release which will upgrade the parser to recognise this syntax.

**Comment by motiz88** [comment]
We've just published v2.99.0 of both `flow-remove-types` and `flow-node`, which should resolve this issue. Future releases will be aligned with Flow releases (e.g. v2.99.0 ↔ Flow 0.99.0) and will always use the latest version of the Flow parser for maximum compatibility.

The code has now [moved](https://github.com/facebook/flow/tree/master/packages/flow-remove-types) to the main Flow repo, and this separate repo will be archived soon; please use the Flow issue tracker for any further discussion.
